commit 9c7d83ae6ba67d6c6199cce24573983db3b56332
Author: Tetsuo Handa <penguin-kernel@I-love.SAKURA.ne.jp>
Date:   Sun Feb 14 12:13:07 2021 +0900

    pstore: Fix warning in pstore_kill_sb()
    
    syzbot is hitting WARN_ON(pstore_sb != sb) at pstore_kill_sb() [1], for the
    assumption that pstore_sb != NULL is wrong because pstore_fill_super() will
    not assign pstore_sb = sb when new_inode() for d_make_root() returned NULL
    (due to memory allocation fault injection).
    
    Since mount_single() calls pstore_kill_sb() when pstore_fill_super()
    failed, pstore_kill_sb() needs to be aware of such failure path.
    
    [1] https://syzkaller.appspot.com/bug?id=6abacb8da5137cb47a416f2bef95719ed60508a0
    
    Reported-by: syzbot <syzbot+d0cf0ad6513e9a1da5df@syzkaller.appspotmail.com>
    Signed-off-by: Tetsuo Handa <penguin-kernel@I-love.SAKURA.ne.jp>
    Signed-off-by: Kees Cook <keescook@chromium.org>
    Link: https://lore.kernel.org/r/20210214031307.57903-1-penguin-kernel@I-love.SAKURA.ne.jp

diff --git a/fs/pstore/inode.c b/fs/pstore/inode.c
index 93a217e4f563..14658b009f1b 100644
--- a/fs/pstore/inode.c
+++ b/fs/pstore/inode.c
@@ -467,7 +467,7 @@ static struct dentry *pstore_mount(struct file_system_type *fs_type,
 static void pstore_kill_sb(struct super_block *sb)
 {
 	mutex_lock(&pstore_sb_lock);
-	WARN_ON(pstore_sb != sb);
+	WARN_ON(pstore_sb && pstore_sb != sb);
 
 	kill_litter_super(sb);
 	pstore_sb = NULL;